File: java/exercises/03-advanced/ex071_lru_cache/LruCache.java
package fewolearning.exercises.advanced.ex071_lru_cache;

import java.util.LinkedHashMap;
import java.util.Map;

/*
Exercise 071 - LRU cache (advanced).

Goal:   Evict the least recently used entry once the cache exceeds its capacity.
Drills: maps, eviction policy, recency tracking.
*/
public final class LruCache<K, V> {
    private final int capacity;
    private final Map<K, V> store;

    public LruCache(int capacity) {
        this.capacity = capacity;
        this.store = new LinkedHashMap<>(16, 0.75f, true);
    }

    public void put(K key, V value) {
        throw new UnsupportedOperationException("TODO");
    }

    public V get(K key) {
        throw new UnsupportedOperationException("TODO");
    }

    public int size() {
        throw new UnsupportedOperationException("TODO");
    }
}

File: php/exercises/03-advanced/ex071_lru_cache/LruCache.php
<?php

declare(strict_types=1);

namespace FeWoLearning\Exercises\Advanced\Ex071LruCache;

/*
Exercise 071 - LRU cache (advanced).

Goal:   Implement a fixed-capacity least-recently-used cache backed by a plain array.
Drills: array insertion order, key re-insertion for recency tracking, array_key_first, eviction.
Passes: LruCacheTest
*/
final class LruCache
{
    private array $store = [];

    public function __construct(private readonly int $capacity)
    {
    }

    public function get(string $key): mixed
    {
        throw new \RuntimeException('TODO');
    }

    public function put(string $key, mixed $value): void
    {
        throw new \RuntimeException('TODO');
    }
}
